mkmapgrids/mkscripgrid.py

```diff
@@ -0,0 +1,140 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri May 31 11:44:57 2024
+
+@author: olgad
+"""
+
+# This script creates a SCRIP grid file for a land-only point or regional case.
+# It is a python version of mkscripgrid.ncl originally written by Erik Kluzek Dec/07/2011
+
+import os
+import numpy as np
+from datetime import datetime
+from netCDF4 import Dataset
+
+def getenv_double(var_name, default):
+    value = os.getenv(var_name)
+    return float(value) if value else default
+
+def getenv_int(var_name, default):
+    value = os.getenv(var_name)
+    return int(value) if value else default
+
+def getenv_str(var_name, default):
+    value = os.getenv(var_name)
+    return value if value else default
+
+def fspan1up(fbegin, fend, number):
+    if number == 1:
+        return np.array([(fbegin + fend) / 2.0])
+    else:
+        return np.linspace(fbegin, fend, number)
+
+# ===========================================================================================================
+# Set a few constants needed later
+cdate = datetime.now().strftime("%y%m%d")
+ldate = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+
+# IMPORTANT NOTE: EDIT THE FOLLOWING TO CUSTOMIZE.
+
+# Input resolution and position
+name = getenv_str("PTNAME", None) # Set your grid name here
+
+latS = getenv_double("S_LAT", 45.5098) # Set the south latitude here
+latN = getenv_double("N_LAT", 45.6098) # Set the north latitude here
+lonE = getenv_double("E_LON", 275.3362) # Set the east longitude here
+lonW = getenv_double("W_LON", 275.2362) # Set the west longitude here
+
+nx = getenv_int("NX", 1) # Set the number of grids along longitude lines here
+ny = getenv_int("NY", 1) # Set the number of grids along latitude lines here
+
+
+imask = getenv_int("IMASK", 1)
```

Comment:
A parser for these options would be nice, so that you do not need to modify the script.